src/main/java/com/google/devtools/build/lib/sandbox/LinuxSandboxedSpawnRunner.java

```diff
@@ -95,11 +110,43 @@ public static boolean isSupported(final CommandEnvironment cmdEnv) throws Interr
     return isSupported;
   }
 
-  private static boolean computeIsSupported(CommandEnvironment cmdEnv, Path linuxSandbox)
+  private static SupportLevel computeIsSupported(CommandEnvironment cmdEnv, Path linuxSandbox)
       throws InterruptedException {
+    try {
+      runBinTrue(cmdEnv, linuxSandbox, /* wrapInBusybox= */ false);
+      return SupportLevel.SUPPORTED;
+    } catch (CommandException e) {
+      if (e instanceof BadExitStatusException badExitStatusException) {
+        var stderr = new String(badExitStatusException.getResult().getStderr(), ISO_8859_1);
+        // Ubuntu 24.04 allows unprivileged user namespaces but denies any permissions that wouldn't
+        // be granted outside the namespace. The first syscall that makes use of extended
+        // permissions is a mount.
+        if (stderr.lines().anyMatch(line -> line.endsWith(": \"mount\": Permission denied"))) {
```

Comment:
What about escaping immediately out instead of nesting.

```suggestion
        if (stderr.lines().noneMatch(line -> line.endsWith(": \"mount\": Permission denied"))) {
          return SupportLevel.NOT_SUPPORTED;
        }
```